tests/PaymentPart/Output/FpdfOutput/FpdfOutputTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Sprain\Tests\SwissQrBill\PaymentPart\Output\FpdfOutput;
+
+use PHPUnit\Framework\TestCase;
+use Sprain\SwissQrBill\PaymentPart\Output\FpdfOutput\FpdfOutput;
+use Sprain\SwissQrBill\PaymentPart\Output\FpdfOutput\Template\QrBillFooter;
+use Sprain\SwissQrBill\QrBill;
+use Sprain\SwissQrBill\QrCode\QrCode;
+use Sprain\Tests\SwissQrBill\TestQrBillCreatorTrait;
+
+class FpdfOutputTest extends TestCase
+{
+    use TestQrBillCreatorTrait;
+
+    /**
+     * @dataProvider validQrBillsProvider
+     * @param string $name
+     * @param QrBill $qrBill
+     */
+    public function testValidQrBills(string $name, QrBill $qrBill)
+    {
+        $variations = [
+            [
+                'printable' => false,
+                'format' => QrCode::FILE_FORMAT_SVG,
+                'file' => __DIR__ . '/../../../TestData/FpdfOutput/' . $name . '.pdf'
+            ],
+            [
+                'printable' => true,
+                'format' => QrCode::FILE_FORMAT_SVG,
+                'file' => __DIR__ . '/../../../TestData/FpdfOutput/' . $name . '.print.pdf'
+            ]
```

Comment:
Interestingly, no print version is generated in tests.

Reply:
This is because Fpdf only accept PNG images. if you use 
`'format' => QrCode::FILE_FORMAT_PNG,`
Files will be created but tests fails anyway.
I am not very experienced with unit testing and I do not know how to validate these tests currently may be you can help to better understand this.